### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by bockwurstballade** [SAP/PyRFC#302] [Q&A] [answered]
Having trouble with the installation. The first question I have whether it is supposed to be called with python2 or python3.

Under both versions, I am getting an error when trying to install

Python 2 output:
`python2 setup.py bdist_wheel
  File "setup.py", line 185
    name=f"{MODULE_NAME}.{MODULE_NAME}",
                                      ^
SyntaxError: invalid syntax
`
Python 3 output:
`python3 setup.py bdist_wheel
Traceback (most recent call last):
  File "setup.py", line 9, in <module>
    from setuptools import setup, find_packages, Extension
ModuleNotFoundError: No module named 'setuptools'
python2 setup.py bdist_wheel
  File "setup.py", line 185
    name=f"{MODULE_NAME}.{MODULE_NAME}",
                                      ^
SyntaxError: invalid syntax
`

**Comment by mpf82** [comment]
Not every combination of Python version and pyrfc version works.

If you're running Python 3.7+, you should be able to use the latest pyrfc (2.7.1)

But judging from the error message you posted, it seems your Python 3 does not support f-strings, so it's probably Python 3.5 or older.

You will have to browse the older pyrfc releases, e.g. https://github.com/SAP/PyRFC/releases/tag/1.9.4 to see whats working for you.

**Comment by bsrdjan** [comment, marked as answer]
Hello @bockwurstballade,

which Python 3 version are you using and on which platform? Python 2 is not supported and supported Python 3 versions are listed here: https://github.com/SAP/PyRFC/blob/main/setup.py#L209
